docs/coding-standards/js.md

```diff
@@ -1,61 +1,168 @@
+```js
+import { nodeListForEach } from '../globals/common'
+
+function Checkboxes ($module) {
+  ...
+}
+
+Checkboxes.prototype.init = function () {
+  ...
+}
+
+export default Checkboxes
+```
+## Comments
+
+Use `/** ... */` for multi-line comments. Include a description, specify types and values for all parameters and return values.
+
+```js
+/**
+* Get the nearest ancestor element of a node that matches a given tag name
+* @param {object} node element
+* @param {string} match tag name (e.g. div)
+* @return {object} ancestor element
+*/
+
+function (node, match) {
+  ...
+  return ancestor
+}
+```
+
+Use `//` for single line comments. Place single line comments on a newline above the subject of the comment.
 
-## Running standard manually
+Use `// FIXME:` to annotate problems.
 
-To check the whole codebase, run:
+Use `// TODO:` to annotate solutions to problems.
+
+## Constructors and methods
+
+Use the prototype design pattern to structure your code.
+
+Create a constructor and define any variables that the object needs
 
-```bash
-npm run lint
+```
+function Checkboxes ($module) {
+  ...
+}
 ```
 
-## Running standard in your editor
+Assign methods to the prototype object. No not overwrite the prototype with a new object as this makes inheritance impossible.
 
-Easier than running standard manually is to install it as a plugin in your editor. This way, it will run automatically while you work, catching errors as they happen on a per-file basis.
+```js
+// bad
+Checkboxes.prototype = {
+  init: function () {
+    ...
+  }
+}
 
-### Sublime Text
+// good
+Checkboxes.prototype.init = function () {
+  ...
+}
+```
 
-Using [Package Control](https://packagecontrol.io/), install [SublimeLinter](http://www.sublimelinter.com/en/latest/) and [SublimeLinter-contrib-standard](https://packagecontrol.io/packages/SublimeLinter-contrib-standard).
+When initialising an object, don’t forget the `new` keyword.
 
-For automatic formatting on save, install [StandardFormat](https://packagecontrol.io/packages/StandardFormat).
+```js
+// bad
+var myCheckbox = Checkbox().init()
 
-### Atom
+//good
+var myCheckbox = new Checkbox().init()
+```
+## Modules
+
+Use ES6 modules (`import`/`export`) over a non-standard module system. You can always transpile to your preferred module system.
+
+```js
+import { nodeListForEach } from '../globals/common'
+...
+export default Checkboxes
+```
 
-Install [linter-js-standard](https://atom.io/packages/linter-js-standard).
+Avoid using wildcard (`import * as nodeListForEach`) imports.
 
-For automatic formatting, install [standard-formatter](https://atom.io/packages/standard-formatter). For snippets, install [standardjs-snippets](https://atom.io/packages/standardjs-snippets).
+Prefer default export over named export.
 
-### Visual Studio Code
+## Polyfilling
```

Comment:
My bad but I think this section is a little messy since I wrote it in chunks and didnt review it all at once, would you be OK with me do a patch?

Reply:
Sure